test/e2e/metrics_e2e_test.go

```diff
@@ -5,59 +5,83 @@ package e2e
-	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"regexp"
 )
 
-var _ = Describe("Metrics", func() {
-	It("endpoint", func() {
-
-		// TestMetrics tests the metrics endpoint of the OLM pod.
-
-		c := newKubeClient(GinkgoT())
-		crc := newCRClient(GinkgoT())
-
-		failingCSV := v1alpha1.ClusterServiceVersion{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       v1alpha1.ClusterServiceVersionKind,
-				APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name: genName("failing-csv-test-"),
-			},
-			Spec: v1alpha1.ClusterServiceVersionSpec{
-				InstallStrategy: v1alpha1.NamedInstallStrategy{
-					StrategyName: v1alpha1.InstallStrategyNameDeployment,
-					StrategySpec: strategy,
-				},
-			},
-		}
-
-		cleanupCSV, err := createCSV(GinkgoT(), c, crc, failingCSV, testNamespace, false, false)
-		Expect(err).ToNot(HaveOccurred())
-		defer cleanupCSV()
+var _ = Describe("Metrics are generated for OLM pod", func() {
 
-		_, err = fetchCSV(GinkgoT(), crc, failingCSV.Name, testNamespace, csvFailedChecker)
-		Expect(err).ToNot(HaveOccurred())
+	var (
+		c   operatorclient.ClientInterface
+		crc versioned.Interface
+	)
+
+	BeforeEach(func() {
+		c = newKubeClient(GinkgoT())
+		crc = newCRClient(GinkgoT())
+
+	})
 
-		// Verify metrics have been emitted for packageserver csv
-		Expect(getMetricsFromPod(c, getOLMPod(c), "8081")).To(And(
-			ContainSubstring("csv_abnormal"),
-			ContainSubstring(fmt.Sprintf("name=\"%s\"", failingCSV.Name)),
-			ContainSubstring("phase=\"Failed\""),
-			ContainSubstring("reason=\"UnsupportedOperatorGroup\""),
-			ContainSubstring("version=\"0.0.0\""),
-			ContainSubstring("csv_succeeded"),
-		))
+	Context("Given an operator group that supports all namespaces", func() {
+		When("a CSV spec does not include Install Mode", func() {
+
+			var (
+				cleanupCSV cleanupFunc
+				failingCSV v1alpha1.ClusterServiceVersion
+			)
+
+			BeforeEach(func() {
+
+				failingCSV = v1alpha1.ClusterServiceVersion{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       v1alpha1.ClusterServiceVersionKind,
+						APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: genName("failing-csv-test-"),
+					},
+					Spec: v1alpha1.ClusterServiceVersionSpec{
+						InstallStrategy: v1alpha1.NamedInstallStrategy{
+							StrategyName: v1alpha1.InstallStrategyNameDeployment,
+							StrategySpec: strategy,
+						},
+					},
+				}
+
+				var err error
+				cleanupCSV, err = createCSV(c, crc, failingCSV, testNamespace, false, false)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = fetchCSV(crc, failingCSV.Name, testNamespace, csvFailedChecker)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				if cleanupCSV != nil {
+					cleanupCSV()
+				}
+			})
+			It("CSV will transition to a failure state with reason UnsupportedOperatorGroup", func() {
```

Comment:
Does this precisely state the desired behavior being tested? This test is verifying that the `csv_abnormal` metric is reported by OLM.